pkg/net/coap/client.go

```diff
@@ -155,42 +144,6 @@ func WithResourceType(in string) OptionFunc {
 	}
 }
 
-func WithInstanceID(in int) OptionFunc {
-	return func(opts message.Options) message.Options {
-		v := "ins=" + strconv.Itoa(in)
-		buf := make([]byte, len(v))
-		opts, _, _ = opts.AddString(buf, message.URIQuery, v)
-		return opts
-	}
-}
-
-func WithDeviceID(in string) OptionFunc {
-	return func(opts message.Options) message.Options {
-		v := "di=" + in
-		buf := make([]byte, len(v))
-		opts, _, _ = opts.AddString(buf, message.URIQuery, v)
-		return opts
-	}
-}
-
-func WithCredentialId(in int) OptionFunc {
-	return func(opts message.Options) message.Options {
-		v := "credid=" + strconv.Itoa(in)
-		buf := make([]byte, len(v))
-		opts, _, _ = opts.AddString(buf, message.URIQuery, v)
-		return opts
-	}
-}
-
-func WithCredentialSubject(in string) OptionFunc {
-	return func(opts message.Options) message.Options {
-		v := "subjectuuid=" + in
-		buf := make([]byte, len(v))
-		opts, _, _ = opts.AddString(buf, message.URIQuery, v)
-		return opts
-	}
-}
```

Comment:
Why you remove this funcs ? Because maybe they are used in plgd/hub or in Kistler

Reply:
I wanted to increase code coverage, it's not used in any plgd repositories. They are easy to implement if we ever need them, however if they are used in Kistler that would be a breaking change. The other option would be to add test cases to cover these functions by tests.

Reply:
Ok you can remove them